### DISCUSSION TITLE
Check for resources outside a scanned region

### DISCUSSION THREAD

**Discussion by ws8n** [prowler-cloud/prowler#2758] [Q&A] [answered]
Hi,

Is it possible to check for resources that are deployed outside my region?
E.g my default region is us-east-1, and I have a VM deployed in US-west-2
I would like to be able to have this flagged out.

Thanks!

**Comment by jfagoagas** [comment, marked as answer]
Hi @ws8n, Prowler is multi-region by default so you don't need to do anything to scan all of them, just `prowler aws`.

In the case you want to input some specific regions you can use the `--region` flag like `prowler aws --region us-east-1 us-west-2`,  please refer to this https://docs.prowler.cloud/en/latest/tutorials/aws/regions-and-partitions/ for more information.

Thanks for using Prowler 🚀

**Comment by ws8n** [comment]
![image](https://github.com/prowler-cloud/prowler/assets/136807879/f632c5bd-8901-4fad-aa71-7785046a6c73)

I scanned without indicating regions, but it seems to be either very slow (running for 90 mins), cuz I wanted to get overall view of all instances in my environment. 

-> Scanning ec2 service |▉▉▉▉▉▉▉▉▉▉▉▋                            | | 84/287 [29%] in 1:25:19
Stuck here for over 1 hour. 
When I killed it, I got disconnected from my ec2 instance, and could not log back in.

**Reply by jfagoagas** [reply]
The scan time depends on how big is your AWS account and some checks looks for secrets in ECS Task Definitions, Lambda Code, Log Groups. If you environment is big I recommend you scanning per region and per service since in a "normal" AWS account the scan time average is 10 minutes.

**Reply by ws8n** [reply]
Is there an option to just scan for EC2 instances? 
Mine is just a default AWS account, but I've created a few EC2 instances in different regions for a POC with Prowler.

**Reply by jfagoagas** [reply]
Hi @ws8n, you can use the `--service ec2` flag to scan just the AWS EC2 service. I recommend you to check the Prowler documentation at https://docs.prowler.cloud/en/latest. If you are still having long audits you can send us the debug log using the `--log-level DEBUG` flag to check it.
